### PR TITLE
Plane: prevent jump to Circle upon Takeoff mode entry unless flying >10s

### PR DESCRIPTION
**Pull request by Hwurzburg**
unfortunately I don't know how to test a hand launch in SITL...and wont be able to test in plane for at least a week or more...

### CODE REVIEW

File: ArduPlane/mode_takeoff.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ void ModeTakeoff::update()
 {
     if (!takeoff_started) {
         // see if we will skip takeoff as already flying
-        if (plane.is_flying() && plane.ahrs.groundspeed() > 3) {
+        if ( (millis() - plane.started_flying_ms > 10000U) && plane.ahrs.groundspeed() > 3) {
```
Comment (rmackay9) [inline ArduPlane/mode_takeoff.cpp]: nitpick: extra space after the bracket. Sorry to be annoying and perhaps not a blocker.

Comment (Hwurzburg) [reply ArduPlane/mode_takeoff.cpp]: fixed

File: ArduPlane/mode_takeoff.cpp [hunk 2]
```diff
@@ -65,7 +65,7 @@ void ModeTakeoff::update()
 {
     if (!takeoff_started) {
         // see if we will skip takeoff as already flying
-        if (plane.is_flying() && plane.ahrs.groundspeed() > 3) {
+        if ((millis() - plane.started_flying_ms > 10000U) && plane.ahrs.groundspeed() > 3) {
```
Comment (peterbarker) [inline ArduPlane/mode_takeoff.cpp]: This doesn't look correct.  You do need to check `is_flying()` as well.

Comment (Hwurzburg) [reply ArduPlane/mode_takeoff.cpp]: added....thought it was reset if not flying...but appears to reset only if flying is restarted

**Comment by tridge** [comment]
thanks for doing this Henry!

**Pull request merged**